src/xpk/commands/cluster_gcluster.py

```diff
@@ -0,0 +1,123 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from ..core.blueprint.blueprint_generator import BlueprintGenerator, BlueprintGeneratorOutput, supported_device_types, a3mega_device_type
+from ..core.docker_manager import DockerManager
+from ..core.gcluster_manager import GclusterManager
+from ..core.core import zone_to_region
+from ..utils.console import xpk_exit, xpk_print
+from ..utils.network import all_IPs_cidr
+from ..utils.file import ensure_directory_exists
+from ..utils.objects import hash_string
+import os
+
+blueprints_path = os.path.abspath('xpkclusters/blueprints')
+gcluster_working_dir = os.path.abspath('xpkclusters/gcluster-out')
+gcloud_cfg_path = os.path.expanduser('~/.config/gcloud')
+
+
+def cluster_create(args) -> None:
+  """Function around cluster creation using Cluster toolkit.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  check_gcloud_authenticated()
+  prepare_directories()
+  gcm = prepare_gcluster_manager()
+  unique_name = get_unique_name(args.project,zone_to_region(args.zone),args.cluster)
+
+  bp = generate_blueprint(blueprint_name=unique_name, args=args)
+  # staging: sending the blueprint file(s) to gcluster's working directory
+  bp_staged_path = gcm.stage_files(blueprint_file=bp.blueprint_file, blueprint_dependencies=bp.blueprint_dependencies)
+  gcm.deploy(blueprint_path=bp_staged_path, deployment_name=unique_name)
+
+  xpk_exit(0)
+
+
+def cluster_delete(args) -> None:
+  """Function around cluster delete for the clusters created by Cluster toolkit.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  check_gcloud_authenticated()
+  prepare_directories()
+  gcm = prepare_gcluster_manager()
+  unique_name = get_unique_name(args.project,zone_to_region(args.zone),args.cluster)
+
+  gcm.destroy_deployment(deployment_name=unique_name)
+
+  xpk_exit(0)
+
+
+def created_by_gcluster(args) -> bool:
+  prepare_directories()
+  unique_name = get_unique_name(args.project,zone_to_region(args.zone),args.cluster)
+  bpg = prepare_blueprint_generator()
+  return bpg.blueprint_exists(unique_name)
+
+
+def get_unique_name(project_id, region, cluster_name):
+  unique_string_hash = hash_string(input_string=f'{project_id}-{region}-{cluster_name}'.lower(), length=5)
```

Comment:
what if cluster name contains _ ?

Reply:
Does that make any problems?

Reply:
won't it collide with charset in hashing function?

Reply:
I wouldn't say so. In the hashstring fuction regardless what the input is, it will be first hashed, and the hashed integer will be convert to base36. that charset is used for converting to base36.